tests/Unit/Booking/BookingsServiceTest.php

```diff
@@ -0,0 +1,363 @@
+use Illuminate\Support\Facades\DB;
+use Tests\TestCase;
+
+class BookingsServiceTest extends TestCase
+{
+    use RefreshDatabase, BookingsTestHelper;
+
+    protected $position;
+    protected $service;
+    private $mockUserModel;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->position = factory(Position::class)->create();
+
+        $this->service = $this->app->make(BookingsServiceInterface::class);
+
+        $this->mockUserModel = User::initModelWithData([
+            'id' => 1234567,
+            'name_fist' => 'First',
+            'name_last' => 'Last',
+            'atcRating' => (object) ['code' => 'S2', 'vatsim_id' => 3],
+        ]);
+    }
+
+    /** @test */
+    public function itDetectsWhenAUserDoesntMeetRatingRequirements()
+    {
+        $specificPosition = factory(Position::class)->create(['callsign' => 'EGGD_APP']);
+
+        $this->assertFalse($this->service->validateRatingRequirement($this->mockUserModel, $specificPosition));
+    }
+
+    /** @test */
+    public function itDetectsWhenAUserMeetsRatingRequirements()
+    {
+        $specificPosition = factory(Position::class)->create(['callsign' => 'EGGD_TWR']);
+
+        $this->assertTrue($this->service->validateRatingRequirement($this->mockUserModel, $specificPosition));
+    }
+
+    /** @test */
+    public function itDetectsWhenBookingOverlapsOnStartTime()
+    {
+        $from = new Carbon('10th January 2018 15:00:00');
+        $to = new Carbon('10th January 2018 16:30:00');
+
+        // create existing booking
+        factory(Booking::class)->create(['from' => $from, 'to' => $to, 'position_id' => $this->position->id]);
+
+        $overlappingFrom = new Carbon('10th January 2019 15:15:00');
+
+        $this->assertFalse($this->service->validateBookingTimes($overlappingFrom, $to, $this->position));
+    }
+
+    /** @test */
+    public function itDetectsWhenBookingOverlapsOnEndTime()
+    {
+
+        // create existing booking
+        factory(Booking::class)->create([
+            'from' => new Carbon('10th January 2019 15:00:00'),
+            'to' => new Carbon('10th January 2019 16:30:00'),
+            'position_id' => $this->position->id,
+        ]);
+
+        $from = new Carbon('10th January 2019 13:15:00');
+        $overlappingTo = new Carbon('10th January 2019 15:15:00');
+
+        $this->assertFalse($this->service->validateBookingTimes($from, $overlappingTo, $this->position));
+    }
+
+    /** @test */
+    public function itDetectsWhenBookingOverlapsCompletely()
+    {
+        $from = new Carbon('10th January 2019 15:00:00');
+        $to = new Carbon('10th January 2019 16:30:00');
+
+        // create existing booking
+        factory(Booking::class)->create([
+            'from' => $from,
+            'to' => $to,
+            'position_id' => $this->position->id,
+        ]);
+
+        $this->assertFalse($this->service->validateBookingTimes($from, $to, $this->position));
```

Comment:
We haven't actually put in a test here to check if we overlap with a gap on each side. I.e. a booking from 15:00 to 16:30, trying to make a booking from 14:00 to 17:00